modules/reactor/src/meshgenerators/PolygonMeshGeneratorBase.C

```diff
@@ -1408,3 +1408,92 @@ PolygonMeshGeneratorBase::biasTermsCalculator(
                 outer_boundary_layer_params.fraction)));
   return biased_terms;
 }
+
+void
+PolygonMeshGeneratorBase::setSectorExtraIDs(MeshBase & mesh,
+                                            const std::string id_name,
+                                            const unsigned int num_sides,
+                                            const std::vector<unsigned int> num_sectors_per_side)
+{
+  const auto extra_id_index = mesh.add_elem_integer(id_name);
+  // vector to store sector ids for each element
+  auto elem_it = mesh.elements_begin();
+  unsigned int id = 1;
+  // starting element id of the current sector
+  for (unsigned int is = 0; is < num_sides; ++is)
+  {
+    // number of elements in the current sector
+    unsigned int nelem_sector =
+        mesh.n_elem() * num_sectors_per_side[is] /
+        (accumulate(num_sectors_per_side.begin(), num_sectors_per_side.end(), 0));
+    // assign sector ids to mesh
+    for (unsigned i = 0; i < nelem_sector; ++i, ++elem_it)
+      (*elem_it)->set_extra_integer(extra_id_index, id);
+    // update sector id
+    ++id;
+  }
+}
+
+void
+PolygonMeshGeneratorBase::setRingExtraIDs(MeshBase & mesh,
+                                          const std::string id_name,
+                                          const unsigned int num_sides,
+                                          const std::vector<unsigned int> num_sectors_per_side,
+                                          const std::vector<Real> ring_radii,
+                                          const std::vector<unsigned int> ring_intervals,
+                                          const bool ring_wise_id,
+                                          const bool quad_center_elements)
+{
+  const auto extra_id_index = mesh.add_elem_integer(id_name);
+  auto elem_it = mesh.elements_begin();
```

Comment:
add a comment that elements are ordered by rings (inner) then by sectors (outer ordering), which is the condition for working with this iterator